src/jllvm/materialization/ByteCodeCompileLayer.cpp

```diff
@@ -1497,6 +1520,14 @@ void codeGenBody(llvm::Function* function, const Code& code, const ClassFile& cl
                     pool.resolve(classFile),
                     [&](const IntegerInfo* integerInfo)
                     { operandStack.push_back(builder.getInt32(integerInfo->value)); },
+                    [&](const StringInfo* stringInfo)
+                    {
+                        llvm::StringRef text = stringInfo->stringValue.resolve(classFile)->text;
+
+                        String* string = stringInterner.intern(text);
+
+                        operandStack.push_back(CppToLLVMType<String*>::getConstant(string, builder));
```

Comment:
Instead of using `CppToLLVMType<String*>::getConstant(string, builder)` its better you use the `builder` and `CreateIntToPtr` to get the object into the IR.

The simple reason being that `CppToLLVMType<T*>` currently returns a pointer type into address space 0 and we need the pointer to be in address space 1 (just use the type returned from `referenceType(builder.getContext())`).
Alternatively, you could define a `CppToLLVMType` specialization for any subtypes of `ObjectInterface` to use `referenceType(builder.getContext())`.

Reply:
done